app/workers/fetch_graph_worker.rb

```diff
@@ -45,6 +50,10 @@ def perform(pid)
     ActiveFedora::SolrService.commit
   end
 
+  def fetch_failed_graph(pid, user)
+    FetchFailedGraphWorker.perform_async(pid, user)
```

Comment:
Are pid and uri the same?
I think this might mean what I wrote initially is almost useless, it only dumps the graph into the cache, it doesn't reindex the work.